stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -398,14 +493,51 @@ public final class _DataStorage {
         }
         
     }
+
+    public func get(_ index: Int) -> UInt8 {
+        switch _backing {
+        case .swift: fallthrough
+        case .immutable: fallthrough
+        case .mutable:
+            return _bytes!.advanced(by: index - _offset).assumingMemoryBound(to: UInt8.self).pointee
+        case .customReference(let d):
+            if d._isCompact() {
+                return d.bytes.advanced(by: index - _offset).assumingMemoryBound(to: UInt8.self).pointee
+            } else {
+                var byte: UInt8 = 0
+                d.enumerateBytes { (ptr, range, stop) in
```

Comment:
It might be more code-size-efficient to just call `getBytes:range:NSMakeRange(offsetIndex, 1)` here. Discontiguous data implementations of that method should do essentially the same thing that you have written here.